dashboard/src/shared/ResourceRef.ts

```diff
@@ -33,9 +35,8 @@ class ResourceRef {
 
   // Creates a new ResourceRef instance from an existing IResource. Provide
   // defaultNamespace to set if the IResource doesn't specify a namespace.
-  // TODO: add support for cluster-scoped resources, or add a ClusterResourceRef
-  // class.
-  constructor(r: IResource, defaultNamespace?: string, defaultFilter?: any) {
+  constructor(r: IResource, cluster: string, defaultNamespace?: string, defaultFilter?: any) {
```

Comment:
for the maintainability of this I would:

 - Make the `defaultNamespace` mandatory. It's easier to understand calls like `new ResourceRef({}, "cluster-a", namespace)`. Also you would notice if you are missing the cluster in some call.
 - Remove the `defaultFilter` param. AFAIK, it's only used in the function `fromCRD` above. You can move it to a setter.

Reply:
> for the maintainability of this I would:

These are both not things I changed, but always keen to improve things...

> 
>     * Make the `defaultNamespace` mandatory. It's easier to understand calls like `new ResourceRef({}, "cluster-a", namespace)`. Also you would notice if you are missing the cluster in some call.

It's not clear to me why we even have this arg... the docstring says in case an IResource doesn't have a namespace, but it's a required field on IResource. I think we are only ever showing namespaced resources? Smells either like a hack to work-around bad input, or old cruft that hasn't been removed? Let me know, but either way I'm keen to change that in a separate branch and move forward here.


> 
>     * Remove the `defaultFilter` param. AFAIK, it's only used in the function `fromCRD` above. You can move it to a setter.

+1. It's a public field, so just set it in `fromCRD`.

Reply:
> It's not clear to me why we even have this arg... the docstring says in case an IResource doesn't have a namespace, but it's a required field on IResource. I think we are only ever showing namespaced resources? Smells either like a hack to work-around bad input, or old cruft that hasn't been removed? Let me know, but either way I'm keen to change that in a separate branch and move forward here.

This parameter is needed because we parse the manifests installed by helm to retrieve the different resources. With helm, it's not necessary to set the namespace. If a resource doesn't have a namespace, it's installed in the namespace of the release.

Reply:
Thanks for the explanation. That being the case, it seems odd to require it as a default even if it's not necessary (rather than just leaving it as an optional default). But I'll take a closer look.